src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -0,0 +1,385 @@
+    /// @param _operatorSignature Contains the operator's signature, salt, and expiry
+    function _registerOperatorWithSig(
+        address _operator,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory _operatorSignature
+    ) internal virtual {
+        if (_operatorRegistered[_operator]) revert OperatorAlreadyRegistered();
+        _operatorRegistered[_operator] = true;
+        _updateOperatorWeight(_operator);
+        IServiceManager(serviceManager).registerOperatorToAVS(_operator, _operatorSignature);
+        emit OperatorRegistered(_operator, serviceManager);
+    }
+
+    /// @notice Updates the weight of an operator and returns the previous and current weights.
+    /// @param _operator The address of the operator to update the weight of.
+    function _updateOperatorWeight(address _operator) internal virtual {
+        int256 delta;
+        uint256 oldWeight;
+        uint256 newWeight;
+        if (!_operatorRegistered[_operator]) {
+            (oldWeight, ) = _operatorWeightHistory[_operator].push(0);
+            delta -= int(oldWeight);
+        } else {
+            newWeight = getOperatorWeight(_operator);
+            (oldWeight, ) = _operatorWeightHistory[_operator].push(newWeight);
+            delta = int256(newWeight) - int256(oldWeight);
+        }
+        _updateTotalWeight(delta);
+        emit OperatorWeightUpdated(_operator, oldWeight, newWeight);
+    }
+
+    /// @dev Internal function to update the total weight of the stake
+    /// @param delta The change in stake applied last total weight
+    /// @return oldTotalWeight The weight before the update
+    /// @return newTotalWeight The updated weight after applying the delta
+    function _updateTotalWeight(
+        int256 delta
+    ) internal returns (uint256 oldTotalWeight, uint256 newTotalWeight) {
+        oldTotalWeight = _totalWeightHistory.latest();
+        int256 newWeight = int256(oldTotalWeight) + delta;
+        newTotalWeight = uint256(newWeight);
+        _totalWeightHistory.push(newTotalWeight);
+        emit TotalWeightUpdated(oldTotalWeight, newTotalWeight);
+    }
+
+    /// @dev Verifies a quorum has:
+    ///     1. Weights that add to 10_000 basis points
+    ///     2. There are no duplicate strategies considered in the quorum
+    /// @param _quorum The new quorum configuration
+    /// @return bool Indicates if the quorum is valid
+    function _isValidQuorum(Quorum memory _quorum) internal pure returns (bool) {
+        StrategyParams[] memory strategies = _quorum.strategies;
+        address lastStrategy;
+        address currentStrategy;
+        uint256 totalMultiplier;
+        for (uint256 i; i < strategies.length; i++) {
+            currentStrategy = address(strategies[i].strategy);
+            if (lastStrategy >= currentStrategy) revert NotSorted();
+            lastStrategy = currentStrategy;
+            totalMultiplier += strategies[i].multiplier;
+        }
+        if (totalMultiplier != BPS) return false;
```

Comment:
hmm, interesting idea. seems odd though that adding strategies would require reducing the multipliers for existing strategies? 
I think this could have negative side effects with asynchronous weight updates to operators, where e.g. the same operator gets a lower weight after a change even though they have the same amount of stake delegated to them

Reply:
Yeah, that's a good point in this construction. I was trying to cache some operations in the update stage and was thinking you would just couple the quorum strategy weight updates with the updateOperators, but it does lead to a discrepancy in your weight if a strategy you were heavily weighted in has its multiplier decreased and you haven't updated the operator to reflect the change.

Reply:
this invariant should address this [97592a6](https://github.com/Layr-Labs/eigenlayer-middleware/pull/194/commits/97592a677513db4c0791741e47014ae58433d30c)

Reply:
seems reasonable. totally avoids what I'm saying as long as the whole operator set is updated at once